jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/GatewaySetupAuthenticationDialog.kt

```diff
@@ -0,0 +1,200 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.gettingstarted
+
+import com.intellij.openapi.observable.properties.PropertyGraph
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.SimpleListCellRenderer
+import com.intellij.ui.components.BrowserLink
+import com.intellij.ui.components.JBTabbedPane
+import com.intellij.ui.dsl.builder.AlignX
+import com.intellij.ui.dsl.builder.bindItem
+import com.intellij.ui.dsl.builder.bindText
+import com.intellij.ui.dsl.builder.panel
+import com.intellij.ui.dsl.builder.toNullableProperty
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.components.BorderLayoutPanel
+import org.jetbrains.annotations.VisibleForTesting
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.jetbrains.core.credentials.loginSso
+import software.aws.toolkits.jetbrains.core.credentials.sono.IDENTITY_CENTER_ROLE_ACCESS_SCOPE
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_REGION
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_URL
+import software.aws.toolkits.jetbrains.core.gettingstarted.SetupAuthenticationNotice.NoticeType
+import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
+import software.aws.toolkits.jetbrains.utils.ui.editorNotificationCompoundBorder
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.CredentialSourceId
+import java.awt.BorderLayout
+import java.util.Optional
+import javax.swing.Action
+import javax.swing.BorderFactory
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+data class GatewaySetupAuthenticationDialogState(
```

Comment:
if this is gateway-only let's move it to `jetbrains-gateway`

Reply:
It seems moving this to `jetbrains-gateway` breaks a few things, and also includes placing a dependency on that package (from the `core` package). This may be a bigger refactor than I originally thought it would be, so I would rather postpone it.